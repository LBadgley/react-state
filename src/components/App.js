import React, { PureComponent } from 'react';
import ColorPicker from './ColorPicker';
import ColorDisplay from './ColorDisplay';

export default class App extends PureComponent {
  state = {
    color: ''
  }

  selectedColor = color => {
    this.setState({ color });
  }

  render() {
    const color = this.state.color;
    return (
    <>
      <ColorPicker selectedColor={this.selectedColor} />
      <ColorDisplay color={color} />
    </>
    );
  }
}
