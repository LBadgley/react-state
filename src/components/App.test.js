import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('app rendering test', () => {
  it('renders the app', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
