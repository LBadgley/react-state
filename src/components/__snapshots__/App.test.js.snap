// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app rendering test renders the app 1`] = `<Incrementor />`;
